here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/query/WhereClauseBuilder.kt

```diff
@@ -115,6 +122,83 @@ class WhereClauseBuilder(private val request: ReadFeatures) {
         }
     }
 
+    private fun encodedGeo(geometry: SpGeometry, geoEncoding: Int): ByteArray? {
+        return PgUtil.encodeGeometry(
+            geometry,
+            Flags().geoGzipOff().geoEncoding(geoEncoding)
+        )
+    }
+
+    private fun resolveTransformation(transformation: SpTransformation, geometry: SpGeometry): String {
+        return when(transformation){
+            is SpBuffer -> resolveBuffer(transformation, geometry)
+            else -> throw NakshaException(
+                NakshaError.UNSUPPORTED_OPERATION,
+                "This transformation is not yet supported: ${transformation::class.simpleName}"
+            )
+        }
+    }
+
+    private fun resolveBuffer(buffer: SpBuffer, geometry: SpGeometry): String {
+        val geo = geoParamForBuffer(buffer, geometry)
+        val distancePlaceholder = placeholderForArg(buffer.distance, PgType.DOUBLE)
+        val bufferStyleParams = bufferStyleParams(buffer)
+        return if(bufferStyleParams != null){
+            "ST_Buffer($geo, $distancePlaceholder, $bufferStyleParams)"
+        } else {
+            "ST_Buffer($geo, $distancePlaceholder)"
+        }
+    }
+
+    private fun geoParamForBuffer(buffer: SpBuffer, geometry: SpGeometry): String {
+        return if(buffer.geography){
+            val wkbPlaceholder = placeholderForArg(
+                encodedGeo(geometry, WKB),
+                PgType.BYTE_ARRAY
+            )
+            "ST_GeogFromWKB($wkbPlaceholder)"
```

Comment:
afaik PostGIS does not have a way to convert TWKB into geography, so this is the only case where wu WKB instead of TWKB

Reply:
You can convert geometry doing `ST_Transform(ST_GeomFromTWKB($1),4326)::geography` (and geometry can be generated from TKWB), this has a significant performance impact only, when the geometry is quite big, for example a polygon or something like this.

Reply:
I was wrong - fixed in new version